### PR TITLE
Issues/dp 287 into issues/dp 281

### PR DESCRIPTION
**Pull request by jason-p-pickering**
Redoing this PR to make it more clear what has changed. Instead of dp-287 going directly into dev, this one is comparing against dp-281.

### CODE REVIEW

**Review by sam-bao — CHANGES_REQUESTED** [review]

File: R/packageSetup.R [hunk 1]
```diff
@@ -86,24 +86,32 @@ headerRow <- function(tool, cop_year = getCurrentCOPYear()) {
...
   if (tool == "Data Pack") {
     if (cop_year == 2020) {
-      schema <- datapackr::cop20_data_pack_schema
+      return(datapackr::cop20_data_pack_schema)
     } else if (cop_year == 2021) {
-      schema <- datapackr::cop21_data_pack_schema
-    } else {schema <- datapackr::data_pack_schema}
+      return(datapackr::cop21_data_pack_schema)
+    } else {
+      return(datapackr::data_pack_schema)
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/packageSetup.R:110]: Capturing this here for a new PR: https://jira.pepfar.net/browse/DP-292

File: tests/testthat/test-package-setup.R [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  expect_error(pick_schema(1999,"OPU Data Pack"))
+ 
+  test_schema<-pick_schema(2020,"Data Pack") 
+  testthat::expect_identical(test_schema,  datapackr::cop20_data_pack_schema)
+  test_schema<-pick_schema(2021,"Data Pack") 
+  testthat::expect_identical(test_schema,  datapackr::cop21_data_pack_schema)
+  
+  #Are we sure we want to return this object for anything other than 2020 
+  # and 2021?
+  test_schema<-pick_schema(1999,"Data Pack")
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply tests/testthat/test-package-setup.R:21]: Capturing here to address in a new PR: https://jira.pepfar.net/browse/DP-292

File: R/createKeychainInfo.R [hunk 3]
```diff
@@ -112,6 +112,8 @@ createKeychainInfo <- function(submission_path = NULL,
   } else if (d$info$tool %in% c("OPU Data Pack", "OPU Data Pack Template")) {
     if (d$info$cop_year == 2020) {
       d$info$schema <- datapackr::cop20OPU_data_pack_schema
+    } else if (d$info$cop_year == 2021) {
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/createKeychainInfo.R:115]: Yes, will address in a new PR, described here: https://jira.pepfar.net/browse/DP-294

File: tests/testthat/test-package-setup.R [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  
+  #Test for a valid country UID
+  test_params<-check_params(country_uids = "JTypsdEUNPw")
+  expect_type(test_params,"list")
+  expect_equal(test_params$country_uids,"JTypsdEUNPw")
+  
+  #Test for an invalid country UID
+  test_params<-check_params(country_uids = "foo")
+  expect_type(test_params,"list")
+  expect_equal(length(check_params(country_uids = "foo")$country_uids),0L)
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply tests/testthat/test-package-setup.R:76]: Yes, running `check_params()` with no parameters, returns an empty list.

File: tests/testthat/test-package-setup.R [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+  #Throw an error if the argument is NULL
+  expect_error(check_params(country_uids = NULL))
+  
+  #Test for a valid COP year
+  test_params<-check_params(cop_year = 2020)
+  expect_type(test_params,"list")
+  expect_equal(test_params$cop_year,2020)
+  
+  #When supplied NULL, return the current COP year
+  #Are we sure about this?
```
Comment (jacksonsj) [inline tests/testthat/test-package-setup.R:87]: For now, yes. We need to review and consolidate our approach to default parameters across the board.

**Review by jacksonsj — APPROVED** [review]

**Pull request merged**